src/TLuaInterpreter.cpp

```diff
@@ -16197,6 +16209,67 @@ int TLuaInterpreter::putHTTP(lua_State* L)
     return 2;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#getHTTP
+int TLuaInterpreter::getHTTP(lua_State* L)
+{
+    auto& host = getHostFromLua(L);
+
+    QString urlString;
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "getHTTP: bad argument #1 type (remote url as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    } else {
+        urlString = QString::fromUtf8(lua_tostring(L, 1));
+    }
+
+    QUrl url = QUrl::fromUserInput(urlString);
```

Comment:
You might want to consider the overloaded form:
`[static] QUrl QUrl::fromUserInput(const QString &userInput, const QString &workingDirectory, QUrl::UserInputResolutionOptions options = DefaultResolution)`
which allows for the handling of a relative path - though what the working directory should be may be subject to further consideration...

Reply:
Exactly – also all the other functions (putHTTP postHTTP …) don't do that either. So if there's a good use case for supplying a default working directory for completing partial URLs here, which I kindof doubt but OK, another patch can replace all of them.